rfcs/helm-oci/README.md

```diff
@@ -0,0 +1,173 @@
+# RFC-xxxx Flux OCI support for Helm
+
+**Status:** provisional
+
+**Creation date:** 2022-03-30
+
+**Last update:** 2022-03-30
+
+## Summary
+
+Given that Helm v3.8 supports [OCI](https://helm.sh/docs/topics/registries/) for package distribution,
+we should extend the Flux Source API to allow fetching Helm charts from container registries.
+
+## Motivation
+
+Helm OCI support is one of the most requested feature in Flux
+as seen on this [issue](https://github.com/fluxcd/source-controller/issues/124).
+
+With OCI support, Flux users can automate chart updates to Git in the same way
+they do today for container images.
+
+### Goals
+
+- Add support for fetching Helm charts stored as OCI artifacts with minimal API changes to Flux.
+- Make it easy for users to switch from HTTP/S Helm repositories to OCI repositories.
+
+### Non-Goals
+
+- Introduce a new API kind for referencing charts stored as OCI artifacts.
+
+## Proposal
+
+Introduce an optional field called `type` to the `HelmRepository` spec.
+
+When not specified, the `spec.type` field defaults to `Default` which preserve the current `HelmRepository` API behaviour.
+
+When the `spec.type` field is set to `OCI`, the `spec.url` field must be prefixed with `oci://` (to follow the Helm conventions).
```

Comment:
Should the type be `OCIRegistry`, rather than just `OCI` - since we want to identify that the URL is reflecting a registry and not just a single image within it?

Reply:
IMO fields can not replace documentation no matter how verbose we make them. The `type` field will be document in code, then it will show up at `kubectl explain helmrepository.spec.type`.